include/fiction/algorithms/simulation/sidb/does_charge_distribution_match_logic_for_input_pattern.hpp

```diff
@@ -0,0 +1,59 @@
+//
+// Created by Jan Drewniok on 12.11.24.
+//
+
+#ifndef FICTION_DOES_CHARGE_DISTRIBUTION_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+#define FICTION_DOES_CHARGE_DISTRIBUTION_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+
+#include <fiction/algorithms/simulation/sidb/detect_bdl_wires.hpp>
+#include <fiction/algorithms/simulation/sidb/is_operational.hpp>
+#include <fiction/technology/charge_distribution_surface.hpp>
+#include <fiction/traits.hpp>
+
+#include <kitty/traits.hpp>
+
+#include <cassert>
+#include <cstdint>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * This function checks if a charge distribution encodes the correct logic for a specified input pattern
+ * based on a provided truth table.
```

Comment:
Can you expand on this docstring, please? I unfortunately don't understand what this function is doing that is not already provided by `is_operational`.